### PR TITLE
*: remove useless code

### PR DESCRIPTION
**Pull request by hawkingrei**
### What problem does this PR solve?

Issue Number: close #xxx

Problem Summary:

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: pkg/ddl/ddl_worker.go [hunk 1]
```diff
@@ -929,23 +929,6 @@ func writeBinlog(binlogCli *pumpcli.PumpsClient, txn kv.Transaction, job *model.
 	}
 }
 
-// waitDependencyJobFinished waits for the dependency-job to be finished.
```
Comment (lance6716) [inline pkg/ddl/ddl_worker.go:932]: can we remove this unused function? @tangenta

**Review by zimulala — APPROVED** [review]

File: pkg/ddl/ddl_worker.go (hunk 1 above)
Comment (zimulala) [reply pkg/ddl/ddl_worker.go:932]: This function is not used at the moment, I think we can be deleted.

**Review by lance6716 — APPROVED** [review]

**Pull request merged**